File: experiments/020-PythonVsJava/cells/java-gradle_1-Empty_free/2022-1c-recuperatorio-parcial1/20260921-204311-76559/analysis/own-tests-project/src/main/java/customerimporter/CUIT.java
package customerimporter;

import java.util.Arrays;
import java.util.List;
import java.util.function.Supplier;

public class CUIT extends Identification {

    public static final String TYPE = "C";

    // "23-2566677-9" size 12, "23-25666777-9" size 13
    private static final int MINIMUM_SIZE = 12;
    private static final int MAXIMUM_SIZE = 13;
    private static final int HEADER_SIZE = 2;
    private static final List<String> VALID_HEADERS = Arrays.asList("20", "23", "24", "25", "26", "27", "30", "33", "34");

    // error messages

    public static String invalidNumberErrorDescription() {
        return "Invalid CUIT number";
    }

    // initialization

    CUIT(String anIdentificationNumber) {
        super(anIdentificationNumber);
        assertIsValidNumber(anIdentificationNumber);
    }

    // identification

    public String type() {
        return TYPE;
    }

    // testing

    public boolean isCUIT() {
        return true;
    }

    public boolean isDNI() {
        return false;
    }

    // numbers

    public Object cuitNumberIfNone(Supplier<Object> aNoneBlock) {
        return number();
    }

    public Object dniNumberIfNone(Supplier<Object> aNoneBlock) {
        return aNoneBlock.get();
    }

    // initialization - private

    private static void assertIsValidNumber(String aNumber) {
        String errorDescription = invalidNumberErrorDescription();

        Validation.assertThat(aNumber.length() >= MINIMUM_SIZE && aNumber.length() <= MAXIMUM_SIZE, errorDescription);
        Validation.assertThat(aNumber.charAt(HEADER_SIZE) == '-' && aNumber.charAt(aNumber.length() - 2) == '-', errorDescription);
        Validation.assertThat(VALID_HEADERS.contains(aNumber.substring(0, HEADER_SIZE)), errorDescription);
        Validation.assertThat(Character.isDigit(aNumber.charAt(aNumber.length() - 1)), errorDescription);
        Validation.assertAreAllDigits(aNumber.substring(HEADER_SIZE + 1, aNumber.length() - 2), errorDescription);
    }
}

File: experiments/020-PythonVsJava/cells/python-pytest_1-Empty_free/2022-1c-recuperatorio-parcial1/20260921-205812-95418/analysis/own-tests-project/src/customerimporter/cuit.py
from customerimporter.identification import Identification


class Cuit(Identification):

    # error messages

    @classmethod
    def invalid_number_error_description(cls):
        return "Invalid CUIT number"

    # type

    @classmethod
    def identification_type(cls):
        return "C"

    # number - validation

    @classmethod
    def assert_is_valid_number(cls, an_identification_number):
        # "23-25666777-9" size 13
        cls.assert_valid_number(len(an_identification_number) >= 12 and len(an_identification_number) <= 13)
        cls.assert_valid_number(an_identification_number[2] == "-" and an_identification_number[-2] == "-")
        cls.assert_valid_number(an_identification_number[:2] in cls.valid_headers())
        cls.assert_valid_number(an_identification_number[-1].isdigit())
        cls.assert_valid_number(an_identification_number[3:-2].isdigit())

    @classmethod
    def valid_headers(cls):
        return ["20", "23", "24", "25", "26", "27", "30", "33", "34"]

    # number

    def cuit_number_if_none(self, a_none_block):
        return self._number

    def dni_number_if_none(self, a_none_block):
        return a_none_block()

    def is_cuit(self):
        return True

    def is_dni(self):
        return False
